server/src/DAL/questionsRepository.ts

```diff
@@ -49,15 +49,25 @@ class DBQuestionsRepository {
 	}
 
 	async updateQuestion(id: models.classes.guid, question: models.interfaces.Question) {
-		let index = this.data?.findIndex(q => q.id === id);
-		if (!this.data || !index || index < 0) {
+		let index = (this.data || []).findIndex(q => q.id === id);
+		if (!this.data || index < 0) {
 			throw new ItemNotInDbError(id, "Question");
 		}
 		this.data[index] = { ...this.data[index], ...question, id };
-		this.writeToFile(this.data);
+		await this.writeToFile(this.data);
 		return this.data[index];
 	}
 
+	async deleteQuestion(id: models.classes.guid) {
+		let index = (this.data || []).findIndex(q => q.id === id);
+		if (!this.data || index < 0) {
```

Comment:
Shouldn't we need to define _findIdex_ method as well as  validation check for _this.data_ and _index_ in 
seperate method?

as you said before - if it repeat more than once the code isn't in his best shape

Reply:
That is a really good idae 🧙🏾‍♂️

Reply:
@ReemCohen Please check out [the fix](https://github.com/ChemiAtlow/YahalomTests/pull/29/commits/04cf1bf1541861ad0e4a1fb91490c2264d7d8ed2)